### ISSUE TITLE
Oracle Linux 7 images missing the kernel-container placeholder RPM

### ISSUE THREAD

**Issue by alexsuter** [oracle/docker-images#1927]
hey guys. currently Oracle 12c fails to build. **Requires: kernel-uek**
do you have any idea?

```
--> Running transaction check
---> Package crontabs.noarch 0:1.11-6.20121102git.el7 will be installed
---> Package dbus-libs.x86_64 1:1.10.24-15.0.1.el7 will be installed
---> Package elfutils-default-yama-scope.noarch 0:0.176-5.el7 will be installed
---> Package json-c.x86_64 0:0.11-4.el7_0 will be installed
---> Package libICE.x86_64 0:1.0.9-9.el7 will be installed
---> Package libSM.x86_64 0:1.2.2-2.el7 will be installed
---> Package libnetfilter_conntrack.x86_64 0:1.0.6-1.el7_3 will be installed
---> Package libnfnetlink.x86_64 0:1.0.1-4.el7 will be installed
---> Package libpath_utils.x86_64 0:0.2.1-32.el7 will be installed
---> Package oracle-database-server-12cR2-preinstall.x86_64 0:1.0-5.el7 will be installed
--> Processing Dependency: kernel-uek for package: oracle-database-server-12cR2-preinstall-1.0-5.el7.x86_64
--> Finished Dependency Resolution
[91mError: Package: oracle-database-server-12cR2-preinstall-1.0-5.el7.x86_64 (ol7_latest)
           Requires: kernel-uek
[0m You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
Removing intermediate container 10fc573e3ecd
The command '/bin/sh -c chmod ug+x $INSTALL_DIR/*.sh &&     sync &&     $INSTALL_DIR/$CHECK_SPACE_FILE &&     $INSTALL_DIR/$SETUP_LINUX_FILE &&     rm -rf $INSTALL_DIR' returned a non-zero code: 1

ERROR: Oracle Database container image was NOT successfully created.
ERROR: Check the output and correct any reported problems with the build operation.
```

**Comment by gvenzl** [comment]
Hey @alexsuter, looks like the OL7 image no longer provides the UEK yum repo file hence `yum` is not possible to pull the uek package.

The build is working fine again with the following quick fix:

1) Inside the 12.2.0.1 directory, create a new file called `uek-ol7.repo` with the following content:
```
[ol7_UEKR5]
name=Latest Unbreakable Enterprise Kernel Release 5 for Oracle Linux $releasever ($basearch)
baseurl=http://yum$ociregion.oracle.com/repo/OracleLinux/OL7/UEKR5/$basearch/
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-oracle
gpgcheck=1
enabled=1
```
2) Add the following (last) line in the Dockerfile:
```
# Copy files needed during both installation and runtime
# -------------
COPY $SETUP_LINUX_FILE $CHECK_SPACE_FILE $INSTALL_DIR/
COPY $RUN_FILE $START_FILE $CREATE_DB_FILE $CONFIG_RSP $PWD_FILE $CHECK_DB_FILE $USER_SCRIPTS_FILE $ORACLE_BASE/
COPY uek-ol7.repo /etc/yum.repos.d/
```

**Comment by Djelibeybi** [comment]
I submitted fixed images to the Official image library: https://github.com/docker-library/official-images/pull/10110 

Once the PR is merged and the images are updated, the builds will start working again.

**Comment by ChristophorusReyhan** [comment]
I get the same issue... I'll wait for the merge.

**Comment by Djelibeybi** [comment]
Note that it takes anywhere from 2-6 hours after the merge for the build systems to replace the official image. More details can be found here: https://github.com/docker-library/faq#an-images-source-changed-in-git-now-what

**Comment by ChristophorusReyhan** [comment]
@Djelibeybi I see, thank you for the info

**Comment by Djelibeybi** [comment]
The fixed image is now available on Docker Hub, so I'm closing this issue. Please open a new issue if you have any further concerns or questions.

**Comment by alexsuter** [comment]
Build is currently running. It seems that it works now. Thanks for this very fast response!

**Comment by ChristophorusReyhan** [comment]
```
---> Package libSM.x86_64 0:1.2.2-2.el7 will be installed
---> Package libnetfilter_conntrack.x86_64 0:1.0.6-1.el7_3 will be installed
---> Package libnfnetlink.x86_64 0:1.0.1-4.el7 will be installed
---> Package libpath_utils.x86_64 0:0.2.1-32.el7 will be installed
---> Package oracle-database-server-12cR2-preinstall.x86_64 0:1.0-5.el7 will be installed
--> Processing Dependency: kernel-uek for package: oracle-database-server-12cR2-preinstall-1.0-5.el7.x86_64
--> Finished Dependency Resolution
Error: Package: oracle-database-server-12cR2-preinstall-1.0-5.el7.x86_64 (ol7_latest)
           Requires: kernel-uek
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
Removing intermediate container 3519e97b935a
The command '/bin/sh -c chmod ug+x $INSTALL_DIR/*.sh &&     sync &&     $INSTALL_DIR/$CHECK_SPACE_FILE &&     $INSTALL_DIR/$SETUP_LINUX_FILE &&     rm -rf $INSTALL_DIR' returned a non-zero code: 1

ERROR: Oracle Database container image was NOT successfully created.
ERROR: Check the output and correct any reported problems with the build operation.
```
Huh... I still get the same issue...

**Comment by Djelibeybi** [comment]
Right. You need to `docker pull oraclelinux:7-slim` image to get the latest build.

**Comment by ChristophorusReyhan** [comment]
It's fine now and the build is currently running. Thanks for the fast response!
